### PR TITLE
Add when() block

### PR DESCRIPTION
**Pull request by boxfoot**
Adds a `when()` block to accept any conditional, as proposed in #242 .

I haven't reimplemented any other blocks (like `env()`) to use this one.

### CODE REVIEW

**Comment by sapegin** [comment]
Cool, thanks!

**Comment by vlad-zhukov** [comment]
There should be a note in the changelog about it.

File: packages/core/lib/when.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Applies an array of webpack blocks only if `condition` is true.
+ *
+ * @param {boolean} condition         Condition to test
+ * @param {Function[]} configSetters  Array of functions as returned by webpack blocks.
+ * @return {Function}
+ */
+function when (condition, configSetters) {
+  assertConfigSetters(configSetters)
+
+  if (condition) {
```
Comment (vlad-zhukov) [inline packages/core/lib/when.js]: Would be nice to have a check above that `condition` is a boolean.

Comment (boxfoot) [reply packages/core/lib/when.js]: I had intentionally *not* done this, on the logic that `condition` could be any truthy or falsy value...

Comment (andywer) [reply packages/core/lib/when.js]: I'm fine with `if (condition)`

Comment (vlad-zhukov) [reply packages/core/lib/when.js]: I am for consistency between code and docs. If we allow any values and cast them to booleans then docs should say that.

**Comment by andywer** [comment]
Applied a minimal change to the README to reflect that truthy conditions will also work. Everyone happy now? :)

**Comment by vlad-zhukov** [comment]
The block wasn't exported in `core/lib/index.js`.

**Comment by andywer** [comment]
@boxfoot Can you please make sure that the `when()` block is actually exported (see `index.js`)?

**Comment by andywer** [comment]
Should be fine to merge now.

**Comment by boxfoot** [comment]
Thanks, @andywer  -- sorry for the slow response, I've just been swamped!

**Comment by andywer** [comment]
No worries, I know the feeling 😉

**Pull request merged**